### ISSUE TITLE
[BUG]: the checker reports accessible name and legend violation on a <fieldset> element

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#1337]
### Project

a11y checker engine

### Browser

Chrome

### Operating system

MacOS

### Description

The group_withInputs_hasName rule reports a violation: Group/Fieldset does not have an accessible name
The WCAG20_Fieldset_HasLegend rule reports a violation: The <fieldset> element does not have a <legend>

The current rule reports a violation if a <fieldset> requires a legend. This is incorrect: (1) per [HTML5 spec](https://html.spec.whatwg.org/multipage/form-elements.html#the-fieldset-element), the <legend> is optional for an <fieldset> element; (2) an <fieldset> element has an implicit ARIA role of 'group', and its accessible name can be fulfilled by aria-label, aria-labelledby, legend, or title.

A similar case is reported on <figure> element, see [#1331 ](https://github.com/IBMa/equal-access/issues/1331) 

### Steps to reproduce

see some examples in test/v2/checker/accessibility/rules/group_withInputs_hasName_ruleunit/test_mixed_1.html

<img width="1304" alt="Screenshot 2023-03-03 at 10 36 02 AM" src="https://user-images.githubusercontent.com/18539498/222776123-9aaecf55-9cd9-4ebb-9771-ddd682722eb3.png">

**Comment by ErickRenteria** [comment]
Tested with: accessibility-checker-engine/test/v2/checker/accessibility/rules/WCAG20_Fieldset_HasLegend_ruleunit/FieldSet-hasEmptyLegend.html
This is still reporting as a violation using the dev-1315 branch. 
@shunguoy please review

**Comment by shunguoy** [comment]
@philljenkins  review

**Comment by ErickRenteria** [comment]
Validated with equal-access/accessibility-checker-engine/test/v2/checker/accessibility/rules/fieldset_legend_valid_ruleunit/FieldSet-hasNoLegend.html
<img width="1161" alt="Screenshot 2023-08-03 at 5 45 49 PM" src="https://github.com/IBMa/equal-access/assets/62436670/e817eaaa-c90f-4cdb-949a-757263946d82">

Validated with test/v2/checker/accessibility/rules/aria_accessiblename_exists_ruleunit/aria_table.html 
<img width="1049" alt="Screenshot 2023-08-03 at 5 54 33 PM" src="https://github.com/IBMa/equal-access/assets/62436670/08747131-40ea-497b-af48-ef5cc91b3498">
